### DISCUSSION TITLE
Missing pthread dependency

### DISCUSSION THREAD

**Discussion by Ziemnono** [jnbrunet/caribou#78] [Q&A] [answered]
I tried to compile the plugin using the following command
`export SOFA_SRC=/opt/sofa_src`
`export SOFA_ROOT=/opt/sofa`
`export SSFF_SRC=/opt/caribou`
`cmake -DCMAKE_PREFIX_PATH=$SOFA_ROOT/lib/cmake -DCMAKE_INSTALL_PREFIX=$SOFA_ROOT/plugins/SofaSimpleForceField -DCMAKE_BUILD_TYPE=Release -S $SSFF_SRC -B $SSFF_SRC/build`

I obtain the following error:
<pre><code>
_BUILD_TYPE=Release -S $SSFF_SRC -B $SSFF_SRC/build
-- Caribou with unittests
-- Caribou with benchmarks
-- Caribou with Python support:
    Version: 3.8.10
    Executable: /usr/bin/python3.8
    Headers: /usr/include/python3.8
    Libraries: /usr/lib/x86_64-linux-gnu/libpython3.8.so
    pybind11 version: 2.6.2
    pybind11 config: /usr/lib/cmake/pybind11/pybind11Config.cmake
-- Caribou with SOFA support
    SOFA version: 21.12.99
    SOFA location: /opt/sofa
-- Caribou without SofaPython (not found)
-- Caribou with OpenMP support
	Libraries: /usr/lib/gcc/x86_64-linux-gnu/9/libgomp.so;/usr/lib/x86_64-linux-gnu/libpthread.so
-- Caribou with MKL support
	Include directories: /usr/include/mkl
	Libraries: -Wl,--start-group /usr/lib/x86_64-linux-gnu/libmkl_gf_lp64.a /usr/lib/x86_64-linux-gnu/libmkl_gnu_thread.a /usr/lib/x86_64-linux-gnu/libmkl_core.a -Wl,--end-group;/usr/lib/gcc/x86_64-linux-gnu/9/libgomp.so;/usr/lib/x86_64-linux-gnu/libpthread.so
	Flags: -m64
-- Caribou.Python.Config: /opt/caribou/build/lib/python3/site-packages/Caribou
-- Caribou.Python.Geometry: /opt/caribou/build/lib/python3/site-packages/Caribou/Geometry
-- Caribou.Python.Topology: /opt/caribou/build/lib/python3/site-packages/Caribou/Topology
-- Configuring done
CMake Error at unittest/Caribou/Geometry/CMakeLists.txt:9 (add_executable):
  Target "Caribou.unittests.Geometry" links to target "Threads::Threads" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at unittest/Caribou/Mechanics/CMakeLists.txt:8 (add_executable):
  Target "Caribou.unittests.Mechanics" links to target "Threads::Threads" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at unittest/Caribou/Topology/CMakeLists.txt:44 (add_executable):
  Target "Caribou.unittests.Topology" links to target "Threads::Threads" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at unittest/SofaCaribou/CMakeLists.txt:25 (add_executable):
  Target "Caribou.unittests.SofaCaribou" links to target "Threads::Threads"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?

CMake Error at unittest/SofaCaribou/CMakeLists.txt:25 (add_executable):
  Target "Caribou.unittests.SofaCaribou" links to target "Threads::Threads"
  but the target was not found.  Perhaps a find_package() call is missing for
  an IMPORTED target, or an ALIAS target is missing?

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
</code></pre>

My settings are the following:
<pre><code>
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal

$ cmake --version
cmake version 3.16.3

$ git log --oneline -n 1
ad34630 (HEAD -> master, origin/master, origin/HEAD) Merge pull request #75 from jnbrunet/add-license-1

$ git log --oneline -n 1
b2a81c713a (HEAD -> master, origin/master, origin/HEAD) [SofaCore] Clearer message when template parameter is not compatible with current context (#2262)

</code></pre>

Thank you for your help and let me know if you need anything else !

**Comment by fredroy** [comment, marked as answer]
Hello,
I guess it came from when we removed the dependency on boost::filesystem in the SOFA repo (which apparently was implicitly finding Threads).
The solution should just be to add find_package(Threads REQUIRED). I will make a PR ASAP.

**Comment by jnbrunet** [comment]
Hey Arnaud,

@fredroy made a [PR](https://github.com/jnbrunet/caribou/pull/79) that should normally fix it. Not sure why the ubuntu 20.04 CI doesn't have an issue with this...

Could you update Caribou on your side and tell us if it worked @Ziemnono ?

Jean-Nico

**Reply by Ziemnono** [reply]
Hi, @jnbrunet and @fredroy,
Thank you for your quick reply! I tried to update the branch but I still have the same error...
Do I miss a dependency? Or shall I update sofa to be up-to-date with the master branch?
Sorry for the trouble...

**Reply by Ziemnono** [reply]
After an update of SOFA, everything worked well, thank you very much everyone!
